pkg/vpc/cleanup.go

```diff
@@ -0,0 +1,78 @@
+package vpc
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha4"
+)
+
+func fmtSecurityGroupNameRegexForCluster(name string) string {
+	const ourSecurityGroupNameRegexFmt = "^eksctl-%s-(cluster|nodegroup)-.+$"
+	return fmt.Sprintf(ourSecurityGroupNameRegexFmt, name)
+}
+
+func findAvailableNetworkInterfaces(provider api.ClusterProvider, spec *api.ClusterConfig) ([]string, error) {
+	input := &ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{&spec.VPC.ID},
+			},
+			{
+				Name:   aws.String("status"),
+				Values: []*string{aws.String("available")},
+			},
+		},
+	}
+	output, err := provider.EC2().DescribeNetworkInterfaces(input)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to list dangling network interfaces in %q", spec.VPC.ID)
+	}
+	securityGroupRE, err := regexp.Compile(fmtSecurityGroupNameRegexForCluster(spec.Metadata.Name))
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to list dangling network interfaces in %q", spec.VPC.ID)
+	}
+	networkInterfaces := []string{}
+	for _, eni := range output.NetworkInterfaces {
+		id := *eni.NetworkInterfaceId
+		for _, sg := range eni.Groups {
+			if securityGroupRE.MatchString(*sg.GroupName) {
+				logger.Debug("will delete %q, as it belongs to our security group %q (%s)", id, *sg.GroupName, *sg.GroupId)
```

Comment:
Super nit: Since this isn't called from deleteNetworkInterfaces(), maybe the debug message shouldn't specify "will delete" but, instead, just something like: "found %q belonging to our security group". As I said, super nit. It just doesn't seem to be part of the "findAvailableNetworkInterfaces()" contract.

Reply:
👍